### DISCUSSION TITLE
I keep crashing my browser when updating state

### DISCUSSION THREAD

**Discussion by ghost** [pmndrs/zustand#202] [General]

**Comment by dai-shi** [comment]
Does it work if you don't _read_ the position in the store?
If so, have you checked this: https://github.com/pmndrs/zustand#transient-updates-for-often-occuring-state-changes

**Reply by dai-shi** [reply]
> if I actually console.log the position data (as it's updating from drag) that doesn't crash either.

It's not clear to me if this is about before updating the store or after.

Would it be possible for you to create a minimal repro in codesandbox?

**Reply by dai-shi** [reply]
Hi, I tried running your example. Confirmed the issue.
Unfortunately, I didn't notice anything explicitly wrong.
The example is still too big and you might want to shave it to find where the root cause is.
I'm not even sure if we can create a repro without r3f.

**Reply by dai-shi** [reply]
You could try the other way around: Create a minimal zustand + r3f app, and then add functionality until the issue arises...

**Reply by dai-shi** [reply]
> What is the difference here from what I did?

The transient updates is something that prevents zustand to update react state and lets developers read store value passively.

tbh, I don't yet understand your example (simply because I'm not used to r3f), so I can't tell the difference. I hope someone familiar to r3f and zustand to help you here.

> Would it be worth trying?

Maybe the priority is to find what is wrong firstly. I'm not sure if the transient updates help your case.
